### DISCUSSION TITLE
how to make  adapter-mongodb  work with TLS enabled clusters

### DISCUSSION THREAD

**Discussion by sourav11b** [nosqlbench/nosqlbench#1378] [Q&A] [answered]
Hi,

i have one-way TLS enabled ie nosqlbench needs to verify the certificate presented by mongodb server. does this work out of box or do i need to make code changes?

**Comment by jshook** [comment, marked as answer]
I think this will require code changes. To streamline using TLS, we have a single "SSL Socket Factory" which is used by all drivers which need one of these from the runtime. This supports both common types of trust configurations. It is at io.nosqlbench.api.engine.util.SSLKsFactory in the nb-api module. If you want to support command-line configurable certificate verification, this is the place to add it.
